x-pack/legacy/plugins/lens/server/usage/collectors.ts

```diff
@@ -74,7 +74,9 @@ function addEvents(prevEvents: Record<string, number>, newEvents: Record<string,
 }
 
 async function isTaskManagerReady(server: Server) {
-  return (await getLatestTaskState(server)) !== null;
+  const result = await getLatestTaskState(server);
+  const runs = get(result, '[0].state.runs', 0);
```

Comment:
You have this same line in a separate file, `const runs = get(result, '[0].state.runs', 0);`, and it seems specific enough to warrant a helper function with a comment. For example, I'm not entirely sure what it's doing, why, or whether or not it's correct, so a function + comment would be helpful.

`isTaskManagerReady` and runs being greater than 0 seem like orthogonal things to me.

Reply:
That's fair, and honestly, this is an open discussion about a _right_ fix for this.

`isTaskManagerReady` could be renamed to something like `isReadyToCollectTelemetry` and that can return what we really care about - if we can now query the task manager and it will return us the telemetry data. We don't want to query the task manager _too soon_ (before the task actually ran), but this idea of _too soon_ is something I just recently observed in testing. Perhaps this is not the root issue?